diff --git a/chrome/installer/linux/debian/postinst b/chrome/installer/linux/debian/postinst
index b284f042a0bdcc9964037ad636bd5e5fa49e7daa..9a87014286de83928ab60139e671f8f02b244207 100755
--- a/chrome/installer/linux/debian/postinst
+++ b/chrome/installer/linux/debian/postinst
@@ -49,8 +49,11 @@ update-alternatives --install /usr/bin/x-www-browser x-www-browser \
 update-alternatives --install /usr/bin/gnome-www-browser gnome-www-browser \
   /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
 
-update-alternatives --install /usr/bin/@@PACKAGE_ORIG@@ @@PACKAGE_ORIG@@ \
-  /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
+if [ '"@@PACKAGE_ORIG@@"' != '"@@USR_BIN_SYMLINK_NAME@@"' ]; then
+  update-alternatives --install /usr/bin/@@PACKAGE_ORIG@@ @@PACKAGE_ORIG@@ \
+    /usr/bin/@@USR_BIN_SYMLINK_NAME@@ $PRIORITY
+fi
+
 
 @@include@@../common/apt.include
 
@@ -63,6 +66,3 @@ if [ ! -e "$DEFAULTS_FILE" ]; then
   echo 'repo_add_once="true"' > "$DEFAULTS_FILE"
   echo 'repo_reenable_on_distupgrade="true"' >> "$DEFAULTS_FILE"
 fi
-
-# Run the cron job immediately to perform repository configuration.
-nohup sh /etc/cron.daily/@@PACKAGE@@ > /dev/null 2>&1 &
